core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -88,89 +85,109 @@ public ClientLogger(String className) {
+        log(INFORMATIONAL_LEVEL, format, args);
     }
 
     /**
-     * Sets the logger to the warning logging level.
+     * Logs a formattable message that uses {@code {}} as the placeholder at {@code warning} log level
+     *
+     * <p><strong>Code Samples</strong></p>
+     * <p>
+     * Logging a message at warning log level
+     * {@codesnippet com.azure.core.implementation.util.clientlogger.warning}
      *
-     * @return Updated ClientLogger if warn is enabled.
+     * @param format The formattable message to log
+     * @param args Arguments for the message, if an exception is being logged last argument is the throwable.
      */
-    public ClientLogger asWarning() {
-        return asLevel(WARNING_LEVEL);
+    public void warning(String format, Object... args) {
+        log(WARNING_LEVEL, format, args);
     }
 
     /**
-     * Sets the logger to the error logging level.
+     * Logs a formattable message that uses {@code {}} as the placeholder at {@code error} log level
      *
-     * @return Updated ClientLogger if error is enabled..
+     * <p><strong>Code Samples</strong></p>
+     * <p>
+     * Logging an error with stack trace
+     * {@codesnippet com.azure.core.implementation.util.clientlogger.error}
+     *
+     * @param format The formattable message to log
+     * @param args Arguments for the message, if an exception is being logged last argument is the throwable.
      */
-    public ClientLogger asError() {
-        return asLevel(ERROR_LEVEL);
+    public void error(String format, Object... args) {
+        log(ERROR_LEVEL, format, args);
     }
 
     /**
-     * Logs a format-able message that uses {@code {}} as the placeholder.
+     * Logs a formattable message that uses {@code {}} as the placeholder at {@code default} log level
      *
      * <p><strong>Code Samples</strong></p>
      * <p>
-     * Logging a message with the default log level
+     * Logging a message at default log level
      * <pre>
      * ClientLogger logger = new ClientLogger(Example.class);
      * logger.log("A message");
      * </pre>
-     * <p>
-     * Logging a format-able warning
-     * <pre>
-     * ClientLogger logger = new ClientLogger(Example.class);
-     * logger.asWarning().log("A format-able message. Hello, {}", name);
-     * </pre>
-     * <p>
-     * Logging an error with stack trace
-     * <pre>
-     * ClientLogger logger = new ClientLogger(Example.class);
-     * try {
-     *    upload(resource);
-     * } catch (Throwable ex) {
-     *    logger.asError().log("Failed to upload {}", resource.name(), ex);
-     * }
-     * </pre>
      *
-     * @param format Format-able message.
-     * @param args   Arguments for the message, if an exception is being logged last argument is the throwable.
+     * @param format The formattable message
+     * @param args Arguments for the message, if an exception is being logged last argument is the throwable.
      */
     public void log(String format, Object... args) {
```

Comment:
Given the changes to logging is this something we would want to keep in this form? Instead would we want to remove the DEFAULT_LOG_LEVEL field and change this method to lookup a configuration value?

Reply:
Yeah, we can remove this method but I am not sure if we have to expose a method to get the configured log level. @JonathanGiles - do you prefer to have a method `getConfiguredLogLevel()`?